Modules/Scripted/VectorToScalarVolume/VectorToScalarVolume.py

```diff
@@ -26,82 +63,420 @@ def __init__(self, parent):
+
+    self.outputSelector = slicer.qMRMLNodeComboBox()
     self.outputSelector.nodeTypes = ["vtkMRMLScalarVolumeNode"]
-    self.outputSelector.setMRMLScene( slicer.mrmlScene )
+    self.outputSelector.hideChildNodeTypes = ["vtkMRMLVectorVolumeNode"]
+    self.outputSelector.setMRMLScene(slicer.mrmlScene)
     self.outputSelector.addEnabled = True
     self.outputSelector.renameEnabled = True
     self.outputSelector.baseName = "Scalar Volume"
-    self.outputFrame.layout().addWidget(self.outputSelector)
+    parametersFormLayout.addRow("Output Scalar Volume: ", self.outputSelector)
 
+    #
+    # Options to extract single components
+    #
+    self.conversionMethodWidget = VectorToScalarVolumeConversionMethodWidget()
+    parametersFormLayout.addRow("Conversion Method: ", self.conversionMethodWidget)
     # Apply button
     self.applyButton = qt.QPushButton("Apply")
     self.applyButton.toolTip = "Run Convert the vector to scalar."
-    self.formLayout.addWidget(self.applyButton)
-    self.applyButton.connect('clicked(bool)', self.onApply)
+    parametersFormLayout.addRow(self.applyButton)
+
+    # Parameter set selector (inspired by SegmentStatistics.py)
+    self.parameterNodeSelector = slicer.qMRMLNodeComboBox()
+    self.parameterNodeSelector.nodeTypes = (("vtkMRMLScriptedModuleNode"), "")
+    self.parameterNodeSelector.addAttribute("vtkMRMLScriptedModuleNode", "ModuleName", "VectorToScalarVolume")
+    self.parameterNodeSelector.selectNodeUponCreation = True
+    self.parameterNodeSelector.addEnabled = True
+    self.parameterNodeSelector.renameEnabled = True
+    self.parameterNodeSelector.removeEnabled = True
+    self.parameterNodeSelector.noneEnabled = False
+    self.parameterNodeSelector.showHidden = True
+    self.parameterNodeSelector.showChildNodeTypes = False
+    self.parameterNodeSelector.baseName = "VectorToScalarVolume"
+    self.parameterNodeSelector.setMRMLScene(slicer.mrmlScene)
+    self.parameterNodeSelector.toolTip = "Pick parameter set"
+    parametersFormLayout.addRow("Parameter set: ", self.parameterNodeSelector)
 
     # Add vertical spacer
     self.layout.addStretch(1)
 
+    # Connections
+    self.parameterNodeSelector.connect('currentNodeChanged(vtkMRMLNode*)', self.setParameterNode)
+
+    # updateParameterNodeFromGui
+    self.inputSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateParameterNodeFromGui)
+    self.outputSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.updateParameterNodeFromGui)
+    self.conversionMethodWidget.methodSelectorComboBox.connect('currentIndexChanged(int)', self.updateParameterNodeFromGui)
+    self.conversionMethodWidget.componentsComboBox.connect('currentIndexChanged(int)', self.updateParameterNodeFromGui)
+
+    self.applyButton.connect('clicked(bool)', self.onApply)
+
+    self.setDefaultParameters()
+
+  def cleanup(self):
+    self.removeObservers()
+
+  def setDefaultParameters(self):
```

Comment:
It should be possible to create a parameter node (with default values) by only using the logic class, so default value initialization should be done in the logic class. See createParameterNode method.

Reply:
Done, moved to logic overriding the createParameterNode.